src/MSRIOGroup.cpp

```diff
@@ -261,39 +291,63 @@ namespace geopm
         }
         std::set<int> cpu_idx = m_platform_topo.domain_nested(GEOPM_DOMAIN_CPU,
                                                               domain_type, domain_idx);
-
+        int rep_cpu = *(cpu_idx.begin());
```

Comment:
What all is going on here with the rep_cpu usage?  Could you explain a bit what the intended functionality is here?

I think rep_cpu for combined signals here means that only 1 of the cores will have it's MSRs read for determining the value that's returned.  Is that right?  I'm curious if that means that if you request temperature at the package level, that it only reads one core?

Reply:
Yes that's my understanding of how this is supposed to work.